doc/advanced-testing-mini-course/src/Lec05/EventQueue.hs

```diff
@@ -33,11 +45,14 @@ fakeEventQueue a0 _clock = do
     enqueue :: IORef Agenda -> Event -> IO ()
     enqueue agenda event = modifyIORef' agenda (push (getEventTime event, event))
 
-    dequeue :: IORef Agenda -> IO Event
-    dequeue agenda = do
+    dequeue :: IORef Agenda -> DequeueTimeout -> IO Event
+    dequeue agenda NoTimeout = do
       a <- readIORef agenda
       case pop a of
         Nothing -> return (CommandEventE Exit)
         Just ((_time, event), a') -> do
           writeIORef agenda a'
           return event
+    dequeue agenda (Timeout micros retry) = do
+      -- advance time?
```

Comment:
I think in the future we can move so that `fakeEventQueue` is setting the clock rather than the event-loop. It is a bit weird that the event-loop does it with the comment.
>  -- This is a noop in production deployment.

But that could be in its own pr.